### PR TITLE
New Metric: n-ary AUC

### PR DESCRIPTION
**Pull request by cle-ros**
- added n-ary AUC metric
- added n-ary AUC test cases

### CODE REVIEW

File: flambe/metric/dev/auc.py [hunk 1]
```diff
@@ -104,3 +123,48 @@ def compute(self, pred: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
         area = np.trapz(tpr, fpr)
 
         return torch.tensor(area / self.max_fpr).float()
+
+
+class NAryAUC(AUC):
```
Comment (jeremyasapp) [inline flambe/metric/dev/auc.py]: rename to MultiClassAUC?
